client/proposal.go

```diff
@@ -549,6 +557,40 @@ func enableVer0Cache(ctx context.Context, c wire.Cacher) {
 	})
 }
 
+func (c *Client) enableVer1Cache() {
+	c.log.Trace("Enabling version 1 cache")
+
+	c.version1Cache.mu.Lock()
+	defer c.version1Cache.mu.Unlock()
+
+	c.version1Cache.enabled++
+}
+
+func (c *Client) releaseVer1Cache() {
+	c.log.Trace("Releasing version 1 cache")
+
+	c.version1Cache.mu.Lock()
+	defer c.version1Cache.mu.Unlock()
+
+	c.version1Cache.enabled--
+	for _, u := range c.version1Cache.cache {
```

Comment:
I thought about it again and i think this is still incorrect when more than one channel is opened.  
Since it releases *all* buffered messages. Imagine:  
```go
enableVer1Cache()
// Version 1 Message is received and buffered
enableVer1Cache()
// Version 1 Message is received and buffered
releaseVer1Cache()
// Now BOTH Messages are replayed instead of just the first.
```

Reply:
Ok i see, this works since you start the handler in a go-routine, otherwise the mutex would already be locked.